src/Shared/Mac/MacKeyChain.cs

```diff
@@ -0,0 +1,303 @@
+        }
+
+        public bool Remove(string service, string account)
+        {
+            IntPtr query = IntPtr.Zero;
+            IntPtr itemRefPtr = IntPtr.Zero;
+            IntPtr servicePtr = IntPtr.Zero;
+            IntPtr accountPtr = IntPtr.Zero;
+
+            try
+            {
+                query = CFDictionaryCreateMutable(
+                    IntPtr.Zero,
+                    0,
+                    IntPtr.Zero, IntPtr.Zero);
+
+                CFDictionaryAddValue(query, kSecClass, kSecClassGenericPassword);
+                CFDictionaryAddValue(query, kSecMatchLimit, kSecMatchLimitOne);
+                CFDictionaryAddValue(query, kSecReturnRef, kCFBooleanTrue);
+
+                if (!string.IsNullOrWhiteSpace(service))
+                {
+                    string fullService = CreateServiceName(service);
+                    servicePtr = CreateCFStringUtf8(fullService);
+                    CFDictionaryAddValue(query, kSecAttrService, servicePtr);
+                }
+
+                if (!string.IsNullOrWhiteSpace(account))
+                {
+                    accountPtr = CreateCFStringUtf8(account);
+                    CFDictionaryAddValue(query, kSecAttrAccount, accountPtr);
+                }
+
+                // Search for the credential to delete and get the SecKeychainItem ref.
+                int searchResult = SecItemCopyMatching(query, out itemRefPtr);
+                switch (searchResult)
+                {
+                case OK:
+                    // Delete the item
+                    ThrowIfError(
+                        SecKeychainItemDelete(itemRefPtr)
+                    );
+                    return true;
+
+                case ErrorSecItemNotFound:
+                    return false;
+
+                default:
+                    ThrowIfError(searchResult);
+                    return false;
+                }
+            }
+            finally
+            {
+                if (query != IntPtr.Zero)
+                    CFRelease(query);
+                if (itemRefPtr != IntPtr.Zero)
+                    CFRelease(itemRefPtr);
+                if (servicePtr != IntPtr.Zero)
+                    CFRelease(servicePtr);
+                if (accountPtr != IntPtr.Zero)
+                    CFRelease(accountPtr);
+            }
+        }
+
+        #endregion
+
+        private static IntPtr CreateCFStringUtf8(string str)
+        {
+            byte[] bytes = Encoding.UTF8.GetBytes(str);
+            return CFStringCreateWithBytes(IntPtr.Zero,
+                bytes, bytes.Length, CFStringEncoding.kCFStringEncodingUTF8, false);
+        }
+
+        private static MacOSKeychainCredential CreateCredentialFromAttributes(IntPtr attributes)
+        {
+            string service = GetStringAttribute(attributes, kSecAttrService);
+            string account = GetStringAttribute(attributes, kSecAttrAccount);
+            string password = GetStringAttribute(attributes, kSecValueData);
```

Comment:
Again, can make this return a `byte[]`:

```suggestion
            byte[] password = GetDataAttribute(attributes, kSecValueData);
```